xarray/core/dataarray.py

```diff
@@ -1501,6 +1501,48 @@ def stack(self, dimensions: Optional[
         ds = self._to_temp_dataset().stack(dimensions, **dimensions_kwargs)
         return self._from_temp_dataset(ds)
 
+    def stack_all(self, name='stacked') -> 'DataArray':
+        """
+        Stack all existing dimensions into a single new dimension.
+
+        New dimension will be added at the end, and the corresponding
+        coordinate variables will be combined into a MultiIndex.
+
+        Parameters
+        ----------
+        name : Name of the new dimension. Optional.
+
+        Returns
+        -------
+        stacked : 1D DataArray with stacked data.
+
+        Examples
+        --------
+
+        >>> arr = DataArray(np.arange(6).reshape(2, 3),
+        ...                 coords=[('x', ['a', 'b']), ('y', [0, 1, 2])])
+        >>> arr
+        <xarray.DataArray (x: 2, y: 3)>
+        array([[0, 1, 2],
+               [3, 4, 5]])
+        Coordinates:
+          * x        (x) |S1 'a' 'b'
+          * y        (y) int64 0 1 2
+        >>> stacked = arr.stack_all()
+        >>> stacked.indexes['z']
```

Comment:
'stacked' instead of 'z'?

I think we should show the repr of `stacked` instead of `stacked.indexes`